dsp/window/window_parametric.go

```diff
@@ -147,31 +147,19 @@ func (v Values) Transform(seq []float64) []float64 {
 	return seq
 }
 
-// ValuesComplex is an arbitrary complex window function.
-type ValuesComplex []complex128
-
-// NewValuesComplex returns a ValuesComplex of length n with weights corresponding
-// to the provided window function.
-func NewValuesComplex(window func([]complex128) []complex128, n int) ValuesComplex {
-	v := make(ValuesComplex, n)
-	for i := range v {
-		v[i] = 1
-	}
-	return window(v)
-}
-
-// Transform applies the weights in the receiver to seq in place, returning the
-// result. If v is nil, Transform is a no-op, otherwise the length of v must
+// TransformComplex applies the weights in the receiver to seq in place, returning
+// the result. If v is nil, Transform is a no-op, otherwise the length of v must
```

Comment:
I don't understand this change, TBH. weighting floats with complex numbers is not a replacement for weighting complex numbers with complex numbers?

Reply:
You're absolutely right; the weighting (I'm sort of reluctant to call it weighting) complex numbers with complex numbers should not have been here. The point of the package is to provide scalar weighting of inputs for Fourier transforms. The approach that was used before was a lazy way to apply scalar weightings (my blame).

Reply:
I get it now. Thanks.